### DISCUSSION TITLE
Request bearer token from api and pass it to another request to get response

### DISCUSSION THREAD

**Discussion by myrulezzz** [grafana/grafana-infinity-datasource#503] [Q&A]
Hi is it possible to send a request to a specific url in infinity to get bearer token and automatically embed it to the next api call to get response with data?
token=$(curl -k --location --request POST 'https://xxxxxxxxx/api/login' --header 'Content-Type: application/x-www-form-urlencoded' --header 'Accept: application/json' --header 'Authorization: Basic xxxxxxxxxxxxxxxxxxxxxxxxxxxxx' --data-urlencode 'email=xxxxxxxxx@xxxxxxxxx.com' --data-urlencode 'password=xxxxxxxxxxxxx' | jq .token)
token=$(echo $token | tr -d '"')
echo Token is $token
curl -k --location --request GET 'https://xxxxxxxxxxxxxxxxxx' --header 'Authorization: Bearer '$token''  | jq . > test.json
This is the request i make with curl

@yesoreyeram any idea?

**Comment by kevineckart** [comment]
would also like to see this

**Comment by KevinDW-Fluxys** [comment]
Yes please, I could really use this for multiple data sources
